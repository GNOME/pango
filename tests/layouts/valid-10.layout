{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "comment" : "Mixed-direction ellipsization",
  "text" : "Hello שלום Γειά σας",
  "font" : "Cantarell 11",
  "justify" : true,
  "ellipsize" : "end",
  "width" : 102400,
  "output" : {
    "is-wrapped" : false,
    "is-ellipsized" : true,
    "unknown-glyphs" : 0,
    "width" : 102400,
    "height" : 19456,
    "log-attrs" : [
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "backspace-deletes-character" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "sentence-boundary" : true,
        "sentence-end" : true,
        "word-boundary" : true
      }
    ],
    "lines" : [
      {
        "start-index" : 0,
        "length" : 30,
        "paragraph-start" : true,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 0,
            "length" : 6,
            "text" : "Hello ",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 76,
                "width" : 11264,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 5
              }
            ]
          },
          {
            "offset" : 6,
            "length" : 8,
            "text" : "שלום",
            "bidi-level" : 1,
            "gravity" : "south",
            "language" : "he",
            "script" : "hebrew",
            "font" : {
              "description" : "DejaVu Sans 11",
              "checksum" : "69ccd07023a72ceb27a5e5c22f728627353b60a198170f5e58dd7014221abf01",
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1332,
                "width" : 10240,
                "is-cluster-start" : true,
                "log-cluster" : 6
              },
              {
                "glyph" : 1324,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 1331,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 1344,
                "width" : 10240,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 14,
            "length" : 1,
            "text" : " ",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "he",
            "script" : "hebrew",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 15,
            "length" : 2,
            "text" : "Γ",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "el",
            "script" : "greek",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 677,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 17,
            "length" : 13,
            "text" : "ειά σας",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "common",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 2,
            "extra-attributes" : [
              {
                "end" : 2147483647,
                "type" : "fallback",
                "value" : false
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1062,
                "width" : 19456,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          }
        ]
      }
    ]
  }
}
