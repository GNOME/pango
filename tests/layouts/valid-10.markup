width=100,justify=true,ellipsize=end
Hello שלום Γειά σας
