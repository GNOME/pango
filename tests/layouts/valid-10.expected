Hello שלום Γειά σας

--- parameters

wrapped: 0
ellipsized: 1
lines: 2
width: 102400

--- attributes

range 0 2147483647

--- directions

0 0 0 0 0 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 12(0) 10(0) 8(0) 14(0) 15(0) 17(0) 19(0) 21(0) 23(0) 24(0) 26(0) 28(0) 28(1) 31(0) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'Hello שלום Γειά σας
'
i=2, index=31, paragraph-start=1, dir=ltr ''

--- runs

i=1, index=0, chars=6, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'Hello '
i=2, index=12, chars=4, level=1, gravity=south, flags=0, font=OMITTED, script=hebrew, language=he, 'שלום'
i=3, index=14, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=hebrew, language=he, ' '
i=4, index=15, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=greek, language=el, 'Γ'
i=5, index=17, chars=7, level=0, gravity=south, flags=2, font=OMITTED, script=common, language=en-us, 'ειά σας'
[0,2147483647]fallback=0
i=6, index=30, no run, line end
i=7, index=31, no run, line end
